src/amo/components/AddonBadges/index.js

```diff
@@ -1,80 +1,109 @@
 import { isQuantumCompatible } from 'core/utils/compatibility';
 import Badge from 'ui/components/Badge';
+import RecommendedBadge from 'ui/components/RecommendedBadge';
+import type { AppState } from 'amo/store';
 import type { AddonType } from 'core/types/addons';
 import type { I18nType } from 'core/types/i18n';
 
 import './styles.scss';
 
 type Props = {|
   addon: AddonType,
+|};
+
+type InternalProps = {|
+  ...Props,
+  _config: typeof config,
+  clientApp: string,
   i18n: I18nType,
 |};
 
-export const AddonBadgesBase = (props: Props) => {
-  const { addon, i18n } = props;
+export class AddonBadgesBase extends React.Component<InternalProps> {
+  static defaultProps = {
+    _config: config,
+  };
 
-  if (!addon) {
-    return null;
-  }
+  render() {
+    const { _config, addon, clientApp, i18n } = this.props;
 
-  const getFeaturedText = (addonType: string) => {
-    switch (addonType) {
-      case ADDON_TYPE_EXTENSION:
-        return i18n.gettext('Featured Extension');
-      case ADDON_TYPE_STATIC_THEME:
-      case ADDON_TYPE_THEME:
-        return i18n.gettext('Featured Theme');
-      default:
-        return i18n.gettext('Featured Add-on');
+    if (!addon) {
+      return null;
     }
-  };
 
-  const isIncompatible =
-    addon.type === ADDON_TYPE_EXTENSION &&
-    isQuantumCompatible({ addon }) === false;
+    const getFeaturedText = (addonType: string) => {
+      switch (addonType) {
+        case ADDON_TYPE_EXTENSION:
+          return i18n.gettext('Featured Extension');
+        case ADDON_TYPE_STATIC_THEME:
+        case ADDON_TYPE_THEME:
+          return i18n.gettext('Featured Theme');
+        default:
+          return i18n.gettext('Featured Add-on');
+      }
+    };
 
-  return (
-    <div className="AddonBadges">
-      {addon.is_featured ? (
-        <Badge type="featured" label={getFeaturedText(addon.type)} />
-      ) : null}
-      {addon.isRestartRequired ? (
-        <Badge
-          type="restart-required"
-          label={i18n.gettext('Restart Required')}
-        />
-      ) : null}
-      {addon.is_experimental ? (
-        <Badge type="experimental" label={i18n.gettext('Experimental')} />
-      ) : null}
-      {isIncompatible ? (
-        <Badge
-          type="not-compatible"
-          label={i18n.gettext('Not compatible with Firefox Quantum')}
-        />
-      ) : null}
-      {addon.requires_payment ? (
-        <Badge
-          type="requires-payment"
-          label={i18n.gettext('Some features may require payment')}
-        />
-      ) : null}
-    </div>
-  );
+    const isIncompatible =
+      addon.type === ADDON_TYPE_EXTENSION &&
+      isQuantumCompatible({ addon }) === false;
+
+    return (
+      <div className="AddonBadges">
+        {_config.get('enableFeatureRecommendedBadges') &&
+        addon.is_recommended &&
+        clientApp !== CLIENT_APP_ANDROID ? (
+          <RecommendedBadge />
+        ) : null}
+        {addon.is_featured ? (
```

Comment:
Shouldn't we hide this badge when `enableFeatureRecommendedBadges` is `true` and we are not on Android?

Reply:
See also: https://github.com/mozilla/addons-frontend/issues/7960

Reply:
Exactly. I was going to defer that to #7960, but if you think I should just do it now I can.

Reply:
I think we could do that here, no?

Reply:
Sure, why not? :-)